### PR TITLE
SLI-1231 Use auth to download sonarlint-omnisharp distributions

### PR DESCRIPTION
**Pull request by damien-urruty-sonarsource**

### CODE REVIEW

File: src/main/java/org/sonarlint/intellij/core/SonarLintEngineFactory.java [hunk 1]
```diff
@@ -143,7 +143,7 @@ private static Map<String, String> prepareExtraProps() {
 
   private static void addOmnisharpServerPaths(SonarLintPlugin plugin, Map<String, String> extraProps) {
     extraProps.put("sonar.cs.internal.omnisharpMonoLocation", plugin.getPath().resolve("omnisharp/mono").toString());
-    extraProps.put("sonar.cs.internal.omnisharpWinLocation", plugin.getPath().resolve("omnisharp/win").toString());
+    extraProps.put("sonar.cs.internal.omnisharpWinLocation", plugin.getPath().resolve("omnisharp/net472").toString());
```
Comment (nquinquenel) [inline src/main/java/org/sonarlint/intellij/core/SonarLintEngineFactory.java:146]: Interesting, what was the impact of this?

File: build.gradle.kts [hunk 2]
```diff
@@ -69,11 +67,6 @@ allprojects {
                 }
             }
         }
-        mavenCentral {
```
Comment (nquinquenel) [inline build.gradle.kts:72]: So we don't rely on mavenCentral at all anymore? Repox is sufficient?

**Review by nquinquenel — APPROVED** [review]
Looks super good!

File: build.gradle.kts (hunk 2 above)
Comment (damien-urruty-sonarsource) [reply build.gradle.kts:72]: Yes. All the dependencies we rely on today are on Repox

File: src/main/java/org/sonarlint/intellij/core/SonarLintEngineFactory.java (hunk 1 above)
Comment (damien-urruty-sonarsource) [reply src/main/java/org/sonarlint/intellij/core/SonarLintEngineFactory.java:146]: I just changed the folder name where the omnisharp files are deployed to. Before it was `win`, now I keep the name of the artifact classifier, which is `net472`

**Pull request merged**